presto-main/src/main/java/io/prestosql/spiller/FileSingleStreamSpiller.java

```diff
@@ -76,13 +77,17 @@ public FileSingleStreamSpiller(
             Path spillPath,
             SpillerStats spillerStats,
             SpillContext spillContext,
-            LocalMemoryContext memoryContext)
+            LocalMemoryContext memoryContext,
+            Optional<SpillCipher> spillCipher)
     {
         this.serde = requireNonNull(serde, "serde is null");
         this.executor = requireNonNull(executor, "executor is null");
         this.spillerStats = requireNonNull(spillerStats, "spillerStats is null");
         this.localSpillContext = spillContext.newLocalSpillContext();
         this.memoryContext = requireNonNull(memoryContext, "memoryContext is null");
+        if (requireNonNull(spillCipher, "spillCipher is null").isPresent()) {
+            closer.register(spillCipher.get()::close);
+        }
```

Comment:
I think you can use `ifPresent`:
```java
requireNonNull(spillCipher, "spillCipher is null")
        .ifPresent(closer::register);
```

Reply:
Sadly I can't without introducing a capturing lambda (plus another lambda) ie:
```java
requireNonNull(spillCipher, "spillCipher is null")
        .ifPresent(cipher -> closer.register(cipher::close));
```
because `SpillCipher` extends `AutoCloseable`, but `Closer.register()` only accepts subclasses of `Closeable`. I know this isn't in the performance sensitive path, but it still seemed easier to just use the conditional form here.